core/templates/dev/head/galleries/gallery.html

```diff
@@ -21,7 +21,12 @@
   <script type="text/javascript">
     GLOBALS.LANGUAGE_CODES_AND_NAMES = JSON.parse(
       '{{LANGUAGE_CODES_AND_NAMES|js_string}}');
-    GLOBALS.CAROUSEL_SLIDES_CONFIG = JSON.parse('{{CAROUSEL_SLIDES_CONFIG|js_string}}');
+    GLOBALS.CAROUSEL_SLIDES_CONFIG = JSON.parse(
+      '{{CAROUSEL_SLIDES_CONFIG|js_string}}');
+    {% if explorations_list %}
```

Comment:
Ah, yep, it's true that an extra back-and-forth is needed. However, we do this extra back-and-forth in every other page, and for the purposes of implementing this particular feature, I think we should keep a similar pattern and not deviate. This also makes things simpler, actually, because if the search query changes, the same (AJAX) action flow can be reused.

In any case, I lean strongly against doing a one-off optimization just for this case, especially since it makes this page behave differently from all the others in the app. If we wanted to do something like what you're proposing, we ought to do it for the app as a whole, and have a separate discussion to understand what it entails. In the meantime, though, this PR should be consistent with existing patterns.